addons/mass_mailing_sms/models/mailing_mailing.py

```diff
@@ -367,3 +369,36 @@ def _get_default_ab_testing_campaign_values(self, values=None):
                 campaign_values['name'] = _("A/B Test: %s", sms_subject)
             campaign_values['ab_testing_sms_winner_selection'] = self.ab_testing_sms_winner_selection
         return campaign_values
+
+    def get_chars_counts(self):
+        """Get character counts for sms widget computation of characters and sms counts.
+
+        :return: Character counts used for links, formatted as `{optOutLink: int, link: int}`.
+        """
+        # opt-out is formatted as "\nSTOP SMS: [base_url]/sms/[mailing_id]/[trace_code]"
+        # links are formatted as "[base_url]/r/[link_tracker_code]/s/[sms_id]"
+        # Note: `url_join` will add a '/' as necessary.
+        if len(self) > 1:
+            return UserError("Call on zero or one record.")
+
+        self.check_access_rights('write')
+
+        counts = {}
+        base_url = self.get_base_url()
+
+        intro = '\n' + self.env['sms.composer']._get_unsubscribe_intro()
+        mailing_id_placeholder = str(
+            self.id if self.id
+            else (self.env['mailing.mailing'].sudo().search([], order="id desc", limit=1).id or 0) + 1
+        )
+        counts['optOutLink'] = len(intro + urljoin(base_url, f'sms/{mailing_id_placeholder}/{self.env["mailing.trace"].CODE_SIZE}'))
+
+        max_sms_id = self.env['sms.sms'].sudo().search([], order="id desc", limit=1)
+        max_sms_id_chars = max(len(str(max_sms_id.id)), 5)  # Assumes a mailing won't be more than 100k sms at once
```

Comment:
Wait, what ? Don't know where this 5 comes from ... is it some "initialize" in case we have currently 3 sms and we expect this number to be actually higher at sending time ? Can't we call "_get_remaining_recipients" (if we have self) ? Maybe costly.